demos/common/hfsm/az_iot_hfsm.h

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file az_iot_hfsm.c
+ * @brief HFSM for Azure IoT Operations.
+ * 
+ * @details Implements fault handling for Device Provisioning + IoT Hub operations
+ */
+#ifndef _az_IOT_HFSM_H
+#define _az_IOT_HFSM_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <azure/az_core.h>
+#include <azure/iot/az_iot_common.h>
+#include "az_hfsm.h"
+
+#ifndef AZ_IOT_HFSM_MIN_RETRY_DELAY_MSEC
+#define AZ_IOT_HFSM_MIN_RETRY_DELAY_MSEC 1000
+#endif 
+
+#ifndef AZ_IOT_HFSM_MAX_RETRY_DELAY_MSEC
+#define AZ_IOT_HFSM_MAX_RETRY_DELAY_MSEC 100000
+#endif 
+
+#ifndef AZ_IOT_HFSM_MAX_RETRY_JITTER_MSEC
+#define AZ_IOT_HFSM_MAX_RETRY_JITTER_MSEC 5000
+#endif 
+
+#define AZ_IOT_HFSM_PROVISIONING_ENABLED
+
+typedef struct
+{
+  az_hfsm hfsm;
```

Comment:
why no underscore on this one?

Reply:
Actually I need to use the _internal pattern from Azure SDKs instead. - here and everywhere else.